server/bin/pbench-sync-satellite.sh

```diff
@@ -266,6 +266,13 @@ if [ -s ${state_change_log} ] ;then
     fi
 fi
 
+if [ $nerrs == 0 ] ; then
+    for entry in "${logdir_for_remote}"/*
+    do
+        rm -rf $entry
```

Comment:
Upon further review of the existing code, I don't think we want to add this blanket recursive `rm` command.

If we review the code, we should see that the `${state_change_log}` is a file that lives in the `${logdir_for_remote}/` directory, and only removed if we were successfully able to remotely change the state on the satellite.  If `pbench-sync-satellite.sh` fails to remotely change the state, we need to leave this file here so that the next execution will attempt to finish before proceeding.

The only other contents of the `${logdir_for_remote}` is the `${logdir}`, which is set to `${logdir_for_remote}/${TS}`.  We already have a `trap` which tries to clean that directory up, so if that is not happening, we need to debug why first.

Reply:
@portante If pbench-sync-satellite.sh fails to change the state on the satellite than the error will be thrown and the removal will not take place. So, there will be no need for executing this script. It happened to be executed if every task is fulfilled successfully.